report_thesis/src/sections/experiments/stacking_ensemble.tex

```diff
@@ -0,0 +1,244 @@
+\subsection{Stacking Ensemble}\label{subsec:stacking_ensemble}
+Given the results of the optimization process, we implemented a stacking ensemble to combine the predictions of the top-performing configurations.
+As with all our experiments, we follow the procedure outlined in Section~\ref{subsec:validation_testing_procedures} to evaluate the performance of the stacking ensemble.
+
+For each oxide, we first identified the top-performing configurations to use in the stacking ensemble.
+To identify the top-performing configurations to use in the stacking ensemble, we developed a category-based method, which categorizes models into distinct groups for systematic comparison.
+This approach aids in selecting models that perform optimally for specific tasks, and ensures diversity in the ensemble.
+We also briefly developed and employed a grid search to find the optimal ensemble of configurations, but time constraints prevented its full implementation.
+Finally, we employed the stacking model and evaluated its performance.
+We present the results as well as the 1:1 plots in Section~\ref{subsec:stacking_ensemble_results}.
+
+\subsubsection{Category Method \& Pipeline Selection}\label{subsec:category_method}
+The category-based method organizes models into the following groups:
+
+\begin{itemize}
+    \item Gradient Boosting: \gls{gbr}, \gls{xgboost}, \gls{ngboost}
+    \item Tree-Based: \gls{etr}, \gls{rf}
+    \item Linear and Regularized Models: \gls{lasso}, Ridge, \gls{enet}
+    \item SVM: \gls{svr}
+    \item PLS: \gls{pls}
+\end{itemize}
+
+We use the top-performing configurations that we identified in Section~\ref{sec:optimization_results} to identify configurations for the stacking ensemble.
+
+For each oxide, we filtered the trial data from the optimization experiment to create a dataset containing configuration information and performance metrics.
+These datasets were sorted by \gls{rmsecv}.
+We then selected unique model types for further analysis, based on the categories above.
+Our process involved selecting a set number of configurations for each category, until a maximum number of configurations for the oxide's ensemble was reached.
+
+To ensure diversity and limit scope, we set a maximum of 1 models per category and 3 models per stacking ensemble, with one ensemble for each oxide.
+For each oxide, we developed pipelines to preprocess the data in accordance with the given configuration, as well as use the models with their optimal hyperparameters.
+These pipelines represent the optimal configurations for each oxide, and are used as base estimators in the stacking ensemble.
+
+\subsubsection{Grid Search}\label{subsec:grid_search}
+To identify the optimal stacking ensemble configuration, we briefly developed and employed a grid search methodology.
```

Comment:
```suggestion
To identify the optimal stacking ensemble configuration, we developed and employed a grid search methodology.
```

Reply:
We already said this earlier on line 8 - reference back instead?

Reply:
```suggestion
```